src/MSBuild/XMake.cs

```diff
@@ -3410,52 +3422,58 @@ private static LoggerDescription ParseLoggingParameter(string parameter, string
                 loggerAssemblyName = loggerAssemblySpec;
             }
 
-            return new LoggerDescription(loggerClassName, loggerAssemblyName, loggerAssemblyFile, loggerParameters, verbosity);
+            return new LoggerDescription(loggerClassName, loggerAssemblyName, loggerAssemblyFile, loggerParameters, verbosity, isOptional);
         }
 
         /// <summary>
         /// Loads a logger from its assembly, instantiates it, and handles errors.
         /// </summary>
         /// <returns>Instantiated logger.</returns>
-        private static ILogger CreateAndConfigureLogger
+        private static bool TryCreateAndConfigureLogger
```

Comment:
I don't like the name `TryCreate` here when it will throw in a not-very-edge case (non-optional logger doesn't exist). I think just `CreateAndConfigureLogger`, but change the return type/out param.